cpp/test/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+cmake_minimum_required (VERSION 3.12)
+
+project (mlt-cpp-test)
+
+add_executable(mlt-cpp-test
+    decode.cpp
+    fastpfor.cpp
+    packed_bitset.cpp
+    util.cpp
+)
+set_property(TARGET mlt-cpp-test PROPERTY CXX_STANDARD 20)
+
+# Tests have access to private headers
+target_include_directories(mlt-cpp-test PUBLIC "${PROJECT_SOURCE_DIR}/../src")
+
+option(MLT_WITH_JSON "Include JSON support" ON)
+if(MLT_WITH_JSON)
+    target_include_directories(mlt-cpp-test PUBLIC "${json_SOURCE_DIR}/include")
+    target_compile_definitions(mlt-cpp-test PUBLIC MLT_WITH_JSON=1)
+endif(MLT_WITH_JSON)
+
+#target_include_directories(mlt-cpp-test PRIVATE "${PROJECT_SOURCE_DIR}/../include")
+
+target_compile_definitions(mlt-cpp-test PRIVATE PROTOZERO_USE_VIEW=std::string_view)
```

Comment:
Maybe this should be set either in the build config or in the source code, but not both?